src/main/java/nl/tudelft/dnainator/ui/DNAinator.java

```diff
@@ -24,20 +32,57 @@
 	private static final int MIN_WIDTH = 300;
 	private static final int MIN_HEIGHT = 150;
 
+	private Stage stage;
+	private BooleanProperty ready = new SimpleBooleanProperty(false);
+
+	private void loadDB() {
+		Task<Graph> task = new Task<Graph>() {
+			@Override
+			protected Graph call() throws Exception {
+				Graph db = Neo4jSingleton.getInstance().getDatabase();
+				ready.set(true);
+				return db;
+			}
+		};
+
+		new Thread(task).start();
+	}
+
+	@Override
+	public void init() {
+		loadDB();
+	}
+
 	@Override
 	public void start(Stage primaryStage) throws Exception {
-		primaryStage.setMinHeight(MIN_HEIGHT);
-		primaryStage.setMinWidth(MIN_WIDTH);
-		primaryStage.setTitle(DNAINATOR);
-		primaryStage.getIcons().add(new Image(getClass().getResourceAsStream(ICON)));
-		primaryStage.setMaximized(true);
-		try {
-			BorderPane rootLayout = FXMLLoader.load(getClass().getResource(FXML));
-			Scene scene = new Scene(rootLayout);
-			primaryStage.setScene(scene);
-		} catch (IOException e) {
-			e.printStackTrace();
+		ready.addListener(event -> {
```

Comment:
Not sure if this can happen, but perhaps you only want to load everything on the switch of the property to `true`.